### ISSUE TITLE
OSX audio input crashes

### ISSUE THREAD

**Issue by ulyssesp** [cinder/Cinder#1519]
The InputAnalyzer sample crashes on start with these two messages:

```
|warning| DeviceRef cinder::audio::DeviceManager::addDevice(const string &)[203] multiple Devices with same key: Built-in Line Output - AppleHDA:1
|error  | void cinder::app::AppBase::executeLaunch()[137] Uncaught exception, type: cinder::audio::AudioDeviceExc, what : Invalid samplerate.
libc++abi.dylib: terminating with uncaught exception of type cinder::audio::AudioDeviceExc: Invalid samplerate.
```

It crashes on https://github.com/cinder/Cinder/blob/master/samples/_audio/InputAnalyzer/src/InputAnalyzerApp.cpp#L58, `mInputDeviceNode >> mMonitorSpectralNode;`.

**Comment by richardeakin** [comment]
What hardware are you using, and what do the Device's samplerates look like in a cinder app? You're going to have to investigate a bit and see why that exception is being thrown. Its likely that you are on a hardware configuration that hasn't been tested yet.

**Comment by ulyssesp** [comment]
Yeah, I did a little investigation. The sample rate was somehow set to 0
instead of the system setting. Changing it manually using setSampleRate
didn't work.

This was using a mono usb webcam mic. It works in windows cinder (same
machine) just fine. After changing to the "line in" input (stereo) on osx
it was no longer crashing.

**Comment by richardeakin** [comment]
What is the webcam mic? What is the name of the Device that is being used by default, and isn't working?  One way to get more information is to run the `[Device::printDevicesToString()]https://github.com/cinder/Cinder/blob/f0414f45d0ef68190e9e8fe720d3c989b62bf4c9/include/cinder/audio/Device.h#L96)` function and list those results here.

I'm afraid there isn't much that I can do without understanding how the incorrect samplerate is being set, so investigation on your side would be much appreciated.
